lib/common/database.dart

```diff
@@ -263,15 +260,12 @@ class DatabaseService {
   }
 
   Future<Task> getTask(String taskID) async {
-    try {
-      DocumentSnapshot<Map<String, dynamic>> taskDocument = await users.doc(userid).collection('tasks').doc(taskID).get();
-      if (taskDocument.exists) {
-        return Task.fromMap(taskDocument.data() ?? {"getTask Error":1}, id: taskDocument.id);
-      }
-    } catch (e) {
-      print("Get Failed");
+    DocumentSnapshot<Map<String, dynamic>> taskDocument = await users.doc(userid).collection('tasks').doc(taskID).get();
+    if (taskDocument.exists) {
+      return Task.fromMap(taskDocument.data() ?? {"getTask Error":1}, id: taskDocument.id);
+    } else {
+      throw Exception("Task not found"); // either way this function should not return a new task if the get fails as that doesnt make sense
```

Comment:
Could we replace the print statement from the original with an exception to catch other errors?

Reply:
we could do that, but if the task isn't found in the db it shouldn't return an empty task as if it found the empty task there

Reply:
Yeah, I'm imagining it not returning anything, just throw exception